base/REPL.jl

```diff
@@ -790,6 +790,11 @@ function setup_interface(repl::LineEditREPL; hascolor = repl.hascolor, extra_rep
             buf = copy(LineEdit.buffer(s))
             edit_insert(buf, input)
             string = takebuf_string(buf)
+            # Parse help and shell commands
+            if string[1] in ('?', ';')
```

Comment:
On doing `length(string) >= 2`, this wouldn't work if someone just pasted `;` or `?`.

About `curspos == 0`:
I assume this is to check if the cursor is at the beginning of the line, and so if something is already there, it shouldn't change the REPL's mode. Since I am calling `repl_keymap[string[1]](s, o)` [here](https://github.com/JuliaLang/julia/pull/15460/files#diff-c626f3e4fe93275a365e661a7096da59R795), it will check the position of the cursor in the definition of `;` or `?` [here](https://github.com/JuliaLang/julia/pull/15460/files#diff-c626f3e4fe93275a365e661a7096da59R777). So, should I add the condition for `curspos == 0` to this?

Thank you for your feedback!